src/App.tsx

```diff
@@ -1,12 +1,72 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { getTodos, createTodo } from './api/todos';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+type Status = 'all' | 'active' | 'completed';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState<Status>('all');
+  const [addTodo, setAddTodo] = useState('');
+
+  /* const loadTodos = async () => {
+    try {
+      const loadedTodos = await getTodos(USER_ID);
+
+      setTodos(loadedTodos);
+    } catch (error) {
+      setErrorMessage('Unable to load todos');
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []); */
+
+  // preffered this one because it's safer
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
+  });
+
+  useEffect(() => {
+    if (!errorMessage) {
+      return;
+    }
+
+    const timerId = setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+
+    // eslint-disable-next-line consistent-return
+    return () => {
+      clearTimeout(timerId);
+    };
+  }, [errorMessage]);
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    createTodo(addTodo)
+      .then(newTodo => {
+        setTodos(todos.concat(newTodo));
+      });
+  }
+
+  let visibleTodos = todos;
+
+  if (status === 'active') {
+    visibleTodos = todos.filter(todo => !todo.completed);
+  }
```

Comment:
let's use `enum + switch/case` for limited number of string values. String values are difficult to maintain in your code